helper/schema/resource_diff.go

```diff
@@ -480,6 +481,67 @@ func (d *ResourceDiff) GetRawConfig() cty.Value {
 	return cty.NullVal(schemaMap(d.schema).CoreConfigSchema().ImpliedType())
 }
 
+// GetRawConfigAt is a helper method for retrieving specific values
+// from the RawConfig returned from GetRawConfig. It returns the cty.Value
+// for a given cty.Path or an error diagnostic if the value at the given path does not exist.
+//
+// GetRawConfigAt is considered advanced functionality, and
+// familiarity with the Terraform protocol is suggested when using it.
+func (d *ResourceDiff) GetRawConfigAt(valPath cty.Path) (cty.Value, diag.Diagnostics) {
```

Comment:
I added a duplicate implementation here for resource diff for usage with `CustomizeDiff` (most likely usage would be running some type of validation that requires knowing about prior state (which can be checked with `(*ResourceDiff).GetRawState`)